python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py

```diff
@@ -120,17 +168,44 @@ def lazy(max_materializations_per_minute: Optional[int] = 1) -> "AutoMaterialize
                 is exceeded, the partitions which would have been materialized will be discarded,
                 and will require manual materialization in order to be updated. Defaults to 1.
         """
+        from dagster._core.definitions.auto_materialize_rule import AutoMaterializeRule
+
         return AutoMaterializePolicy(
-            on_missing=False,
-            on_new_parent_data=False,
-            for_freshness=True,
+            rules={
+                AutoMaterializeRule.materialize_on_required_for_freshness(),
+                AutoMaterializeRule.skip_on_parent_outdated(),
+            },
             max_materializations_per_minute=check.opt_int_param(
                 max_materializations_per_minute, "max_materializations_per_minute"
             ),
         )
 
+    @public
+    def without_rules(self, *rules_to_remove: "AutoMaterializeRule") -> "AutoMaterializePolicy":
```

Comment:
@schrockn do you generally believe in avoiding `*args` parameters like this? I think I recall you had a good reason for it, but I'm forgetting what it is.

Reply:
Well seems like a good idea to default to not using it, as it limits your ability to add parameters in backwards compat way in the future

Reply:
I think in this case, additional parameters could be added as keyword arguments, similar to https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/definitions/asset_selection.py?L107. Won't protest too hard, just think it's a bit nicer to do:

```python
my_policy.with_rules(rule1, rule2)
```

vs

```python
my_policy.with_rules({rule1, rule2})
```

Reply:
I don't feel strongly